File: src/backend/api/src/main/kotlin/ru/durchschlagen/models/Auction.kt
package ru.durchschlagen.models

import kotlinx.serialization.Serializable

enum class AuctionStatus {
    ACTIVE,
    CLOSE
}

@Serializable
data class Auction(
    val id: Int,
    var status: AuctionStatus?,
    var title: String?,
    var lot_id: Int,
    var owner_id: Int,
    var finish: String?,
    var cost: Int,
)

File: src/frontend/site/src/domain/Auction.ts
import Comment from './Comment'

type Auction = {
  id: number
  status: 'ACTIVE' | 'CLOSE'
  owner: string
  title: string
  end: string
  cost: number
  lot: {
    name: string
  }
  comments: Comment[]
}

export default Auction
